### PR TITLE
Fixes guarantees around {Chart,ChartOptions}.userOptions being immutable.

### PR DESCRIPTION
**Pull request by alantreadway**
Ensures that we don't mutate `userOptions`, so that subsequent `Chart.updateDelta()` calls are working against the previously provided options, not our mutated/cleaned copy.

Additional fixes:
- Fixes the cleaning of `Chart.queuedIserOptions` that was obscuring the main behaviour.
- Consolidates `optionsModule.ts` tests into `optionsModule.test.ts`.

### CODE REVIEW

File: packages/ag-charts-community/src/module/optionsModule.ts [hunk 1]
```diff
@@ -114,7 +111,7 @@ export class ChartOptions<T extends AgChartOptions = AgChartOptions> {
             this.processedOptions.legend?.enabled == null
         ) {
             this.processedOptions.legend ??= {};
-            this.processedOptions.legend.enabled = this.processedOptions.series!.length > 1;
+            this.processedOptions.legend.enabled = (this.processedOptions.series?.length ?? 0) > 1;
```
Comment (iMoses) [inline packages/ag-charts-community/src/module/optionsModule.ts]: IIRC at this stage `series` is guaranteed to be an array thanks to `sanityCheckAndCleanup`

File: packages/ag-charts-community/src/module/optionsModule.ts [hunk 2]
```diff
@@ -214,7 +211,7 @@ export class ChartOptions<T extends AgChartOptions = AgChartOptions> {
             userPalette,
         };
 
-        const processedSeries = options.series!.map((series) => {
+        const processedSeries = options.series?.map((series) => {
```
Comment (iMoses) [inline packages/ag-charts-community/src/module/optionsModule.ts]: ditto

**Review by iMoses — APPROVED** [review]
LGTM 👍 
just minor notes about probably unnecessary type changes.

File: packages/ag-charts-community/src/module/optionsModule.ts (hunk 1 above)
Comment (alantreadway) [reply packages/ag-charts-community/src/module/optionsModule.ts]: Reverted, there was a need for this during testing, but rolling back I don't see a problem anymore.

File: packages/ag-charts-community/src/module/optionsModule.ts (hunk 2 above)
Comment (alantreadway) [reply packages/ag-charts-community/src/module/optionsModule.ts]: Reverted

**Pull request merged**